tensilelite/Tensile/KernelWriterAssembly.py

```diff
@@ -8368,16 +8406,38 @@ def localWriteSwapOffsets(self, kernel, internalPointerSwap, tP):
     module = Module("localWriteSwapOffsets")
     if needSwap:
       #fixme-iui  need to use wrapping increment for double or triple buffering:
-      if internalPointerSwap:
-        tP["localWriteSwapByteOffset"] = 0 if tP["localWriteSwapByteOffset"] else kernel["LdsOffsetA_Blk"]
-        module.addComment1("(EPS=1) local write swap internal offset -> %u" % tP["localWriteSwapByteOffset"])
-      else:
-        if kernel["LocalWriteUseSgpr%s"%tc]:
-          module.add(SXorB32(
+      if internalPointerSwap or kernel["StoreSwapAddr"]:
+
+        if not kernel["StoreSwapAddr"]:
+          if kernel["LocalWriteUseSgpr%s"%tc]:
```

Comment:
I suggest to unify LocalWriteUseSgpr case code.
No need to duplicate almost same code at 3 locations.
Only difference is src0 for StoreSwapAddr case.
We just need to switch src0 for StoreSwapAddr.

By the way, LocalWriteUseSgpr code did not exist in internalPointerSwap before your change.
Was this missing before your change?

Reply:
Good point, refactored.

> By the way, LocalWriteUseSgpr code did not exist in internalPointerSwap before your change.
Was this missing before your change?

No, I added this to fix a test failure - but I don't think its actually needed, removed. I think the initial test failure was resolved when adding additional checks for `numVgprLocal{Read,Write}Addr`